internal/api/rest_client.go

```diff
@@ -63,16 +60,17 @@ func (c restClient) DoWithContext(ctx context.Context, method string, path strin
 	if err != nil {
 		return err
 	}
-	defer resp.Body.Close()
 
 	success := resp.StatusCode >= 200 && resp.StatusCode < 300
 	if !success {
+		defer resp.Body.Close()
 		return api.HandleHTTPError(resp)
 	}
 
 	if resp.StatusCode == http.StatusNoContent {
 		return nil
 	}
+	defer resp.Body.Close()
```

Comment:
If we move lines 70-72 above line 64 we can remove one of the `defer resp.Body.Close()` lines.

Reply:
In case of HTTP 204 No Content, `resp.Body` will be nil. This is why I'm closing the body in two different places